Code/Engine/Core/ResourceManager/Implementation/ResourceManager_inl.h

```diff
@@ -53,48 +53,64 @@ ezTypedResourceHandle<ResourceType> ezResourceManager::GetExistingResource(ezStr
 
 template <typename ResourceType, typename DescriptorType>
 ezTypedResourceHandle<ResourceType> ezResourceManager::CreateResource(ezStringView sResourceID, DescriptorType&& descriptor, ezStringView sResourceDescription)
+{
+  return CreateResourceInternal<ResourceType, DescriptorType>(sResourceID, std::move(descriptor), sResourceDescription, false);
+}
+
+template <typename ResourceType, typename DescriptorType>
+ezTypedResourceHandle<ResourceType> ezResourceManager::CreateResourceInternal(ezStringView sResourceID, DescriptorType&& descriptor, ezStringView sResourceDescription, bool bAllowGetFallback)
 {
   static_assert(std::is_rvalue_reference<DescriptorType&&>::value, "Please std::move the descriptor into this function");
 
+  ezTypedResourceHandle<ResourceType> hResource;
+  ResourceType* pResource = nullptr;
   EZ_LOG_BLOCK("ezResourceManager::CreateResource", sResourceID);
+  {
+    EZ_LOCK(s_ResourceMutex);
+    // In this locked scope, we decide whether we are creating the resource or waiting for it to be created by someone else.
+    if (bAllowGetFallback)
+      hResource = GetExistingResource<ResourceType>(sResourceID);
```

Comment:
I don't understand the purpose of this fallback resource here, why and how is this used for creating a resource?

Reply:
So I had to combine the code from `ezResourceManager::GetOrCreateResource` and `ezResourceManager::CreateResource` as they effectively do the same except for this little change which is this bool: `bAllowGetFallback`. If set, it first tries to `get` the resource, i.e. fall back to an already existing resource.